index.js

```diff
@@ -1,41 +1,53 @@
 var FeedParser = require('feedparser');
-var request = require('request');
+var axios = require('axios');
+
+
 
 module.exports = function fetch(url, callback) {
-  var req = request(url);
-  var feedparser = new FeedParser();
-  var items = [];
-  var errorRaised = false;
-
-  req.setHeader('user-agent', 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/31.0.1650.63 Safari/537.36');
-  req.setHeader('accept', 'text/html,application/xhtml+xml');
-
-  req.on('error', _raiseError);
-  req.on('response', function(res) {
-    if (res.statusCode != 200) return this.emit('error', new Error('Bad status code'));
-    var encoding = res.headers['content-encoding'] || 'identity';
-    var charset = getParams(res.headers['content-type'] || '').charset;
-    res = maybeDecompress(res, encoding);
-    res.pipe(feedparser);
-  });
+  
 
-  feedparser.on('error', _raiseError);
-  feedparser.on('end', function() { if(!errorRaised) callback(null, items); });
-  feedparser.on('readable', function() {
-    var post;
-    while (post = this.read()) {
-      items.push(post);
+  axios.get(url, {
+    responseType: 'stream',
+    headers: {
+      'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_8_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/31.0.1650.63 Safari/537.36',
+      'Accept': 'text/html,application/xhtml+xml',
+      'Accept-Encoding': 'gzip, deflate'
     }
-  });
+  }).then(response => {
+    if (response.status !== 200) {
+      throw new Error('Bad status code');
+    }
+    const feedparser = new FeedParser();
 
-  function _raiseError(error) {
-    if (!errorRaised) callback(error);
-    errorRaised = true;
-  }
+    const items = [];
+    let errorRaised = false;
+    // response.data.pipe(maybeDecompress(response.data, response.headers['Content-Encoding'])).pipe(feedparser);
+
+    feedparser.on('error', _raiseError);
```

Comment:
Not sure but I think this `feedparser` event handler maybe should be out of the promise. It seems weird to define a event handler just in case a Promise fullfill

Reply:
I see it weird too. Reviewing the `feedparser` [documentation](https://www.npmjs.com/package/feedparser), it seems to utilize fetch instead Axios. I will try to refactor the code and use it by looking the example